spk/tt-rss/src/service-setup.sh

```diff
@@ -26,145 +28,298 @@ SVC_KEEP_LOG=y
-    fi
-    # Restore the configuration file
-    ${CP} "${TMP_DIR}/${PACKAGE}/config.php" "${WEB_DIR}/${PACKAGE}/config.php"
-    SPK_REV=$(cat "${TMP_DIR}/${PACKAGE}/${VERSION_FILE}")
-    if [ "${SPK_REV}" -lt "14" ]
-    then
-      # Parse old configuration and save to new config format
-      sed -i -e "s|define('DB_TYPE', '\(.*\)');|putenv('TTRSS_DB_TYPE=\1');|" \
-        -e "s|define('DB_HOST', '\(.*\)');|putenv('TTRSS_DB_HOST=\1');|" \
-        -e "s|define('DB_USER', '\(.*\)');|putenv('TTRSS_DB_USER=\1');|" \
-        -e "s|define('DB_NAME', '\(.*\)');|putenv('TTRSS_DB_NAME=\1');|" \
-        -e "s|define('DB_PASS', '\(.*\)');|putenv('TTRSS_DB_PASS=\1');|" \
-        -e "s|define('SINGLE_USER_MODE', \(.*\));|putenv('TTRSS_SINGLE_USER_MODE=\1');|" \
-        -e "s|define('SELF_URL_PATH', '\(.*\)');|putenv('TTRSS_SELF_URL_PATH=\1');|" \
-        -e "s|define('DB_PORT', '\(.*\)');|putenv('TTRSS_DB_PORT=\1');|" \
-        -e "s|define('PHP_EXECUTABLE', \(.*\));||" \
-        "${WEB_DIR}/${PACKAGE}/config.php"
-      echo "putenv('TTRSS_PHP_EXECUTABLE=${PHP}');">>"${WEB_DIR}/${PACKAGE}/config.php"
-    fi
-    if [ "${SPK_REV}" -lt "15" ]
-    then
-      sed -i -e "s|putenv('TTRSS_DB_PASS=.*');|putenv('TTRSS_DB_PASS=${wizard_mysql_password_ttrss}');|" \
-        "${WEB_DIR}/${PACKAGE}/config.php"
-      echo "putenv('TTRSS_MYSQL_DB_SOCKET=/run/mysqld/mysqld10.sock');">>"${WEB_DIR}/${PACKAGE}/config.php"
-    fi
+  if [ "${SYNOPKG_DSM_VERSION_MAJOR}" -ge 7 ]; then
+    touch "${SYNOPKG_PKGVAR}/.dsm7_migrated"
+  fi
+  # Restore the configuration file
+  ${CP} "${TMP_DIR}/${PACKAGE}/config.php" "${WEB_DIR}/${PACKAGE}/config.php"
+  SPK_REV=$(cat "${TMP_DIR}/${PACKAGE}/${VERSION_FILE}")
+  if [ "${SPK_REV}" -lt "14" ]
+  then
+    # Parse old configuration and save to new config format
+    sed -i -e "s|define('DB_TYPE', '\(.*\)');|putenv('TTRSS_DB_TYPE=\1');|" \
+      -e "s|define('DB_HOST', '\(.*\)');|putenv('TTRSS_DB_HOST=\1');|" \
+      -e "s|define('DB_USER', '\(.*\)');|putenv('TTRSS_DB_USER=\1');|" \
+      -e "s|define('DB_NAME', '\(.*\)');|putenv('TTRSS_DB_NAME=\1');|" \
+      -e "s|define('DB_PASS', '\(.*\)');|putenv('TTRSS_DB_PASS=\1');|" \
+      -e "s|define('SINGLE_USER_MODE', \(.*\));|putenv('TTRSS_SINGLE_USER_MODE=\1');|" \
+      -e "s|define('SELF_URL_PATH', '\(.*\)');|putenv('TTRSS_SELF_URL_PATH=\1');|" \
+      -e "s|define('DB_PORT', '\(.*\)');|putenv('TTRSS_DB_PORT=\1');|" \
+      -e "s|define('PHP_EXECUTABLE', \(.*\));||" \
+      "${WEB_DIR}/${PACKAGE}/config.php"
+    echo "putenv('TTRSS_PHP_EXECUTABLE=${PHP}');">>"${WEB_DIR}/${PACKAGE}/config.php"
+  fi
+  if [ "${SPK_REV}" -lt "15" ]
+  then
+    sed -i -e "s|putenv('TTRSS_DB_PASS=.*');|putenv('TTRSS_DB_PASS=${wizard_mysql_password_ttrss}');|" \
+      "${WEB_DIR}/${PACKAGE}/config.php"
+    echo "putenv('TTRSS_MYSQL_DB_SOCKET=/run/mysqld/mysqld10.sock');">>"${WEB_DIR}/${PACKAGE}/config.php"
+  fi
+  # Check config file for legacy PHP exec to migrate
```

Comment:
You could restrict this check to only installation older than the one being published

Reply:
I'm not sure i understand. This appears in a code block which checks SPK_REV < 15. Is it that this condition is changed? Do you prefer that I use something like the SYNOPKG_OLD_PKGVER variable to check the package version instead?

Reply:
I mean that any existing installation (16 at least) would be using the virtual-php binary.
Either you ensure the setting is updated for those or you add a basic redirection.

Reply:
I think I get you now, the quoted lines were 299-302 and you are probably referring to the lines 302 onwards for the check for the virtual-php binary. The way it is constructed currently is not based on package version. It will check for any version the contents of the `config.php` file to see if the line for the variable `TTRSS_PHP_EXECUTABLE` exists. If it does, it checks that it matches the declared PHP binary. If not, it will replace the line to match.

Reply:
@smaarn, for consistency with your previous logic, I've adjusted it. Let me know if this works for you.